sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -1298,9 +1298,9 @@ class GpuStringSplitMeta(
       if (str != null) {
         if (RegexParser.isRegExpString(str.toString)) {
           willNotWorkOnGpu("regular expressions are not supported yet")
-        }
-        if (str.numChars() == 0) {
-          willNotWorkOnGpu("An empty regex is not supported yet")
+          if (str.numChars() == 0) {
```

Comment:
Why did this change? Do we have tests for split on empty String that will make sure that this is covered?

Reply:
This is a bug. String splitting can be done with an empty delimiter. In addition, an empty string should not be a regex-it is just a literal string.
According to string splitting API, empty delimiter means splitting by whitespace.

Reply:
I just checked again and realized that the splitting behavior of libcudf and Java for empty delimiter is inconsistent. For empty delimiter, Java splits the input into an array of individual chars while libcudf splits by whitespace.

Then we indeed can't support empty delimiter.

Reply:
I just filed an issue for it: https://github.com/rapidsai/cudf/issues/10251